### DISCUSSION TITLE
ChakraUI + NextJs app router cause Warning: Extra attributes from the server

### DISCUSSION THREAD

**Discussion by skorphil** [chakra-ui/chakra-ui#8348] [Q&A] [answered]
Hi, i discovered that ColorModeScript causing

    Warning: Extra attributes from the server: class
    at body
    at html
    at NotFoundErrorBoundary

I have it in root layout of my app

    // layout.jsx
    export default function RootLayout({ newrecord, children }) {return (
    <html lang="en">  
    <body style={{ height: "100dvh" }}>  
    <ColorModeScript initialColorMode={theme.config.initialColorMode} /> // this is causing warning

Please do you know any way to fix this?

PS
Without ColorModeScript i get huge lag between theme switching from default to dark at the page loading. ColorMode script solving this
https://chakra-ui.com/docs/styled-system/color-mode#for-nextjs BUT there is only pages router example
Here i described the problem with theme changing https://github.com/skorphil/nextjs-form/issues/8#issuecomment-1911889349

**Comment by Vijaykv5** [comment]
Hi @skorphil 

I think One possible solution is to move the `<ColorModeScript>` component to the <Head> section of your layout, which should prevent it from being rendered as an extra attribute in the <body> tag.

**Reply by skorphil** [reply]
Thanks @Vijaykv5
What y mean by moving to the section of my layout? It is already in my layout.jsx file

**Comment by segunadebayo** [comment, marked as answer]
`ColorModeScript` is v2. v2 is maintenance-only.

In v3, wrap with `ColorModeProvider` (`next-themes`) and set `suppressHydrationWarning` on `<html>`. Do not put a color-mode script inside `<body>`.
